### PR TITLE
Hw multithreading

### PR DESCRIPTION
**Pull request by Ilya-brinchuk**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -9,8 +9,64 @@
...
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 
 </project>
```
Comment (oleksandrdziubiuk) [inline pom.xml]: не має нового рядка

File: src/main/java/MyRunnable.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            try {
+                String data = Thread.currentThread().getName() + " value = "
+                        + counter.getCount() + "\n";
+                Files.write(Paths.get(PATH_FILE), data.getBytes(), StandardOpenOption.APPEND);
+                counter.increment();
+                logger.info(data);
+            } catch (IOException e) {
+                throw new RuntimeException("Can't write in file by this path: " + PATH_FILE, e);
+            }
+
```
Comment (oleksandrdziubiuk) [inline src/main/java/MyRunnable.java]: лишній пробіл

File: src/main/java/MyThread.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            try {
+                String data = Thread.currentThread().getName() + " value = "
+                        + counter.getCount() + "\n";
+                Files.write(Paths.get(PATH_FILE), data.getBytes(), StandardOpenOption.APPEND);
+                counter.increment();
+                logger.info(data);
+            } catch (IOException e) {
+                throw new RuntimeException("Can't write in file by this path: " + PATH_FILE, e);
+            }
+
```
Comment (oleksandrdziubiuk) [inline src/main/java/MyThread.java]: так само

**Review by oleksandrdziubiuk — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -9,8 +9,62 @@
...
+                    <linkXRef>false</linkXRef>
+                </configuration>
+            </plugin>
+        </plugins>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-war-plugin</artifactId>
+                    <version>3.3.1</version>
```
Comment (RomanDubovskyi) [inline pom.xml]: do we need war plugin?

File: src/main/java/Main.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+public class Main {
+    public static void main(String[] args) {
+        Counter counter = new Counter();
+        MyRunnable myRunnable = new MyRunnable(counter);
+        MyThread myThread = new MyThread(counter);
+        new Thread(myThread).start();
```
Comment (RomanDubovskyi) [inline src/main/java/Main.java]:
```suggestion
        MyThread myThread = new MyThread(counter);
        myThread.start();
```

File: src/main/java/MyRunnable.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import org.apache.log4j.Logger;
+
+public class MyRunnable implements Runnable {
+    private static final Logger logger = Logger.getLogger(MyRunnable.class);
+    private static final String PATH_FILE = "src/main/resources/test.txt";
```
Comment (RomanDubovskyi) [inline src/main/java/MyRunnable.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(MyRunnable.class);
```
we don't need this if we have logger

File: src/main/java/Counter.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+public class Counter {
+    private int count;
+
+    public int getCount() {
+        return count;
+    }
+
+    public void increment() {
+        count++;
```
Comment (RomanDubovskyi) [inline src/main/java/Counter.java]:
```suggestion
    public int increment() {
      return  count++;
```
that'll make the code shorter

File: src/main/java/MyRunnable.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        while (counter.getCount() < 100) {
+            try {
+                String data = Thread.currentThread().getName() + " value = "
+                        + counter.getCount() + "\n";
+                Files.write(Paths.get(PATH_FILE), data.getBytes(), StandardOpenOption.APPEND);
```
Comment (RomanDubovskyi) [inline src/main/java/MyRunnable.java]: make the logger write data to file
```suggestion
```

File: src/main/java/MyThread.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    private static final Logger logger = Logger.getLogger(MyThread.class);
+    private static final String PATH_FILE = "src/main/resources/test.txt";
+    private Counter counter;
+
+    public MyThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
```
Comment (RomanDubovskyi) [inline src/main/java/MyThread.java:12]: same here

File: src/main/resources/log-file.log [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+ 17:46:37,507 INFO  MyRunnable:23 - Thread-2 value = 95
+
+ 17:46:37,508 INFO  MyThread:23 - Thread-1 value = 96
+
+ 17:46:37,510 INFO  MyRunnable:23 - Thread-2 value = 98
+
+ 17:46:37,510 INFO  MyThread:23 - Thread-1 value = 98
+
+
```
Comment (RomanDubovskyi) [inline src/main/resources/log-file.log]: missing empty line at the end of the file

File: src/main/resources/test.txt [hunk 11]
```diff
@@ -0,0 +1,100 @@
+Thread-2 value = 0
+Thread-1 value = 0
```
Comment (RomanDubovskyi) [inline src/main/resources/test.txt]: one file is enough, we don;t need this one

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**